src/compute/src/render/reduce.rs

```diff
@@ -1189,8 +1205,13 @@ where
             });
         to_aggregate.push(collection);
     }
-    let collection =
-        differential_dataflow::collection::concatenate(&mut collection.scope(), to_aggregate);
+
+    // now concatenate, if necessary, multiple aggregations
+    let collection = if to_aggregate.len() == 1 {
+        to_aggregate[0].clone()
```

Comment:
You can avoid the `clone` by calling `remove` or `swap_remove` on the vector. `remove` is probably a bit more intuitive.

Reply:
Thanks! I have changed this to `remove` now.